### PR TITLE
[plugin-mobile-app] Add step to perform double tap on element

### PR DESCRIPTION
**Pull request by IBrun**

### CODE REVIEW

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java [hunk 1]
```diff
@@ -111,6 +113,22 @@ public void tap(WebElement element, Duration duration)
...
+     * Performs double tap on the <b>element</b>
+     * See details for
+     * <a href="https://github.com/appium/appium-xcuitest-driver#mobile-doubletap">iOS</a>,
+     * <a href="https://github.com/appium/appium-uiautomator2-driver/blob/master/docs/android-mobile-gestures.md#mobile-doubleclickgesture">Android</a>
+     * @param element element to tap, must not be {@code null}
+     */
+    //CHECKSTYLE:ON
+    public void doubleTap(WebElement element)
+    {
+        String script = genericWebDriverManager.isAndroid() ? "mobile: doubleClickGesture" : "mobile: doubleTap";
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java:127]: what about tvOS?

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/steps/TouchSteps.java [hunk 2]
```diff
@@ -94,6 +94,17 @@ public void tapByLocator(Locator locator)
...
 
+
+    /**
+     * Performs double tap on <b>element</b> located by <b>locator</b>
+     * @param locator locator to find an element
+     */
+    @When("I double tap on element located by `$locator`")
+    public void doubleTapByLocator(Locator locator)
+    {
+        findElementToTap(locator).ifPresent(touchActions::doubleTap);
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/steps/TouchSteps.java:105]: the validation message in `findElementToTap` may confuse users, it refers to simple `tap`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java [hunk 3]
```diff
@@ -111,6 +113,34 @@ public void tap(WebElement element, Duration duration)
...
+        else if (genericWebDriverManager.isIOS())
+        {
+            script = Optional.of("mobile: doubleTap");
+        }
+        script.ifPresent(s ->
+        {
+            Map<String, Object> args = Map.of("elementId",
+                    WebDriverUtils.unwrap(element, RemoteWebElement.class).getId());
+            webDriverProvider.getUnwrapped(JavascriptExecutor.class).executeScript(s, args);
+        });
```
Comment (uarlouski) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java:141]: the action does nothing if the script is empty, should not we throw unsupported platform here?

File: vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java [hunk 4]
```diff
@@ -111,6 +113,34 @@ public void tap(WebElement element, Duration duration)
...
+        {
+            script = "mobile: doubleClickGesture";
+        }
+        else if (genericWebDriverManager.isIOS())
+        {
+            script = "mobile: doubleTap";
+        }
+        else
+        {
+            throw new IllegalArgumentException("The functionality is supported only for Android and iOS platforms");
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/mobileapp/action/TouchActions.java:138]:
```suggestion
            throw new IllegalArgumentException("Double tap action is available only for Android and iOS platforms");
```

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**